src/services/auth.service.ts

```diff
@@ -0,0 +1,50 @@
+import { instance } from '../api/axios.api';
+import {
+  IResetPasswordRequest,
+  IUserRegistration,
+  IResponseUserData,
+  IResetPassword,
+  IUserData,
+  IUserId,
+  IUser,
+} from '../types/types';
+
+export const AuthService = {
+  async registration(
+    userData: IUserRegistration,
+  ): Promise<IResponseUserData | undefined> {
+    const { data } = await instance.post<IResponseUserData>(
+      'auth/sign-up',
+      userData,
+    );
+    return data;
+  },
+
+  async login(userData: IUserData): Promise<IUser | undefined> {
+    const { data } = await instance.post<IUser>('auth/sign-in', userData);
```

Comment:
Error Handling: It's important to handle errors that might occur during API requests. Consider adding try-catch blocks to catch any potential errors and handle them appropriately.

Reply:
Done